zc_events/backends.py

```diff
@@ -0,0 +1,227 @@
+import ujson as json
+import pika
+import pika_pool as pika_pool_lib
+import redis
+import traceback
+from zc_events.config import settings
+from zc_events.request import Request
+from zc_events.responses import Response
+
+_DEFAULT_ROUTING_KEY = ''
+_LOW_PRIORITY = 0
+_HIGH_PRIORITY = 9
+
+
+def _noop(request):
+    pass
+
+
+def dispatch_task(name, data):
+    """Dispatch the task for processing on the server side.
+
+    Example:
+        @app.task('microservice.events')
+        def listener(event_name, data):
+            from zc_events.backends import dispatch_task
+            return dispatch_task(event_name, data)
+
+
+    Note:
+        This function relies up `JOB_MAPPING` to be defined in your settings,
+        which is a dict with the key corresponding to a name and the value being
+        a function which will accept an `Request` paramater. If no name is found,
+        nothing happens with the request.
+
+        This function also needs the `RPC_BACKEND` setting to be instantiated
+        so it can use the `respond` method.
+
+    Args:
+        name (str): The name of the function to be called.
+        data (dict or None): The data to be used to populate the Request object.
+    """
+    request = Request(data)
+    func = settings.JOB_MAPPING.get(name, _noop)
+    try:
+        response = {
+            'data': func(request),
+            'has_errors': False,
+            'errors': []
+        }
+    except Exception as e:
+        response = {
```

Comment:
I suggest adding some logging here.  For existing code that doesn't look for a response, there's presently no good way to detect exceptions.